backend/EkoFunkcje/EkoFunkcje/Features/Attachments/AddAttachmentFunction.cs

```diff
@@ -16,14 +16,24 @@ public static class AddAttachmentFunction
         [FunctionName("AddAttachment")]
         public static async Task<HttpResponseMessage> Run(
             [HttpTrigger(AuthorizationLevel.Function, "post", Route = "interventions/{interventionId}/attachments")] HttpRequestMessage req,
-            [Blob("attachments/{interventionId}/{rand-guid}", FileAccess.Write , Connection = Config.StorageConnectionName)] CloudBlockBlob newBlob,
+            IBinder binder, string interventionId,
             ILogger log)
         {
             HttpContentHeaders contentHeaders = req.Content.Headers;
             if (contentHeaders.ContentLength == 0) {
                 return req.CreateErrorResponse(HttpStatusCode.BadRequest, "No attachment sent");
             }
 
+            var fileName = req.Content.Headers.ContentDisposition?.FileName;
+            if (fileName == null)
+            {
+                return req.CreateErrorResponse(HttpStatusCode.BadRequest, "No FileName ContentDisposition");
+            }
+            var attribute = new BlobAttribute($"attachments/{interventionId}/{fileName}", FileAccess.Read);
```

Comment:
Wbrew poczatkowemu załżeniu, nie możemy zapisywać pliku pod file name. Dlatego zostawiłem rand-guid.
Problem był taki, że jeśli nazwa pliku zawierała np. spację albo inne znaki które nie są alfanumeryczne, to ten znak był enkodowany i potem generowaly się złe linki na froncie.

Stąd moje rozwiązanie, żeby scieżką do pliku był dalej GUID (nie ma problemu z enkodowaniem i wadliwymi linkami), ale przypisujemy do bloba "Properties.ContentType" i "Properties.ContentDisposition", czyli zachowujemy wszystkie informacje. Potem przy pytaniu o listę te informacje zwracamy w AttachmentDTO

Reply:
W takim razie zamykam tego PR. Poprawki do adresów Google API utworzę osobno